### PR TITLE
[fosscuda/2019a] PyTorch 1.2.0 and magma 2.5.1

### PR DESCRIPTION
**Pull request by branfosj**
I've split this out from ~~#8753~~. Patches are in ~~#8753~~ which is now merged.

### CODE REVIEW

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.2.0-fosscuda-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    ('Pillow', '6.0.0'),
+]
+
+download_dep_fail = True
+
+prebuildopts = 'PYTORCH_BUILD_VERSION=%(version)s PYTORCH_BUILD_NUMBER=1 VERBOSE=1 '
+prebuildopts += 'LDFLAGS="$LDFLAGS -ldl" '
+prebuildopts += 'USE_FFMPEG=ON USE_GLOO_IBVERBS=1 USE_GFLAGS=ON USE_GLOG=ON '
+prebuildopts += 'CUDNN_LIB_DIR=$EBROOTCUDNN/lib64 CUDNN_INCLUDE_DIR=$EBROOTCUDNN/include '
+prebuildopts += 'TORCH_CUDA_ARCH_LIST="3.5 6.0 7.0" '
```
Comment (smoors) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-1.2.0-fosscuda-2019a-Python-3.7.2.eb]: @branfosj why limit to only these 3 cuda archs here?
In PyTorch-1.0.1-fosscuda-2018b-Python-3.6.6.eb we have: `TORCH_CUDA_ARCH_LIST="3.0 3.2 3.5 3.7 5.0 6.0 6.1 7.0 7.2"`
In the Dockerfile of v1.2.0: `TORCH_CUDA_ARCH_LIST="3.5 5.2 6.0 6.1 7.0+PTX"`

**Comment by boegel** [comment]
@branfosj This needs to be updated to include the patch that was added in ~~#8986~~?

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
bear-pg0212u15b.bear.cluster - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/71d97d21c1aa4ef62e38c94e32d502d1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3300.joltik.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, Python 3.6.8
See https://gist.github.com/b5671b09037496ccd3e8756358376918 for a full test report.

**Comment by boegel** [comment]
close/re-open to re-trigger Travis (and make it pick up latest `develop` for patch from #9010)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node153.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/9cd51d708c7858b5fb7e28ae01658a59 for a full test report.

**Comment by boegel** [comment]
I was hoping to get a test report from another type of system first before merging this, finally got that to work: last report is from an Intel Ivy Bridge node with a Tesla K20x GPU...

Thanks *a lot* for your efforts on this @branfosj, remind me to buy you a beer on the next opportunity...

**Comment by boegel** [comment]
Going in, thanks @branfosj!

**Comment by branfosj** [comment]
:)
Thanks with the help with the bug reports and testing.

**Pull request merged**